troubleshooting_101.rst

```diff
@@ -94,12 +178,57 @@ without proper permissions (in Linux, ports < 1024 are privileged):
     
 Permission denied reading to / writing from disk
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-
+File system permissions, and attributes (chattr)?
 
 
 Out of disk space
 ^^^^^^^^^^^^^^^^^
-(finding large files, and also finding deleted-but-open files)
+
+If you have appropriate monitoring in place, hopefully you are troubleshooting
+a full disk because you received a proactive alert. If not, take an action item
+to dive deep into :doc:`monitoring_101` and configure your notification systems to
+alert for any volume that nears 75% utilization and configure a second critical
+alert if it exceeds 90% utilization.
+
+A frequent cause of downtime is a volumes disk space being fully utilizes.
+Operations engineers often find themselves logged in to a system not knowing what
+is specifically wrong but through habit have learned one of the first things
+they will want to do is check disk space utilization on all of the volumes.
+
+.. code-block:: console
+    
+    user@opsschool ~$ df -h
+    Filesystem            Size  Used Avail Use% Mounted on
+    /dev/sda6             454G  454G    0G 100% /
+    tmpfs                  12G     0   12G   0% /lib/init/rw
+    udev                   12G  144K   12G   1% /dev
+    tmpfs                  12G     0   12G   0% /dev/shm
+    /dev/sda1             236M   41M  183M  19% /boot
+
+Here we can see that /dev/sda6 is 100% used. We better quickly find the
+culprit.
+
+Find files that are larger than 50M:
+
+.. code-block:: console
+
+    user@opsschool ~$ sudo find / -size +50M
+    ..
+    user@opsschool ~$ sudo find / -size +50M -exec ls -lh {} \; | awk '{ print $9 ": " $5 }'
+
+
+Often times, you are going to find a large log file. In fact, you may quickly
+find out you are now solving for two or three symptoms. For example,
+the application servers disk filled, because it was logging a new critical
+event, because database schema changed.
+
+NOTE: Be very careful, just because you can use a few of these tools to
+identify large files, it does not mean they are the culprit or that they
+should be deleted. For example, it would be disastorous to delete a 1.1G file
+named /var/lib/mysql/ibdata1 or /var/lib/mysql/mysql-bin.000971. Use caution
+and if you aren't sure what the file is, be sure to Google it and identify the
+proper way to perform maintenance.
+
```

Comment:
This is really solid.
If you have some time, pulling in some content from Adam Jacob's Velocity 2013 talk would be really good.
He talks about the OODA (observe, orient, decide, act) loop, and how it applies to ops.
The slides are here: http://adamhjk.github.io/good-at-ops/#/6/14 (press 'down' to scroll through that section).

We also have a section for postmortems, but it's currently empty. We can link to it here either now, or when we have more content.